app/models/proposal.rb

```diff
@@ -9,15 +9,21 @@ class Proposal < ActiveRecord::Base
       event :approve, :transitions_to => :approved
       event :reject, :transitions_to => :rejected
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
     end
     state :approved do
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
     end
     state :rejected do
       # partial approvals and rejections can't break out of this state
       event :partial_approve, :transitions_to => :rejected
       event :reject, :transitions_to => :rejected
       event :restart, :transitions_to => :pending
+      event :cancel, :transitions_to => :cancelled
+    end
+    state :cancelled do
+      event :partial_approve, :transitions_to => :cancelled
```

Comment:
At this point, we are not allowing them to restart a cancelled request, although it might be nice to look at that soon